File: python/nbody/body.py
import numpy as np
from numpy.typing import NDArray

class Body:
    def __init__(self, mass: float, position: NDArray[np.float64], velocity: NDArray[np.float64]) -> None:
        self.mass = mass
        self.position = np.array(position, dtype=float)
        self.velocity = np.array(velocity, dtype=float)
        self.force = np.zeros(2, dtype=float)

    def reset_force(self) -> None:
        self.force[:] = 0.0

File: rust/src/body.rs
#[derive(Clone, Debug)]
pub struct Body {
    pub mass: f64,
    pub position: [f64; 2],
    pub velocity: [f64; 2],
    pub force: [f64; 2]
}

impl Body {
    pub fn new(mass: f64, position: [f64; 2], velocity: [f64; 2]) -> Self {
        Self {
            mass,
            position,
            velocity,
            force: [0.0; 2]
        }
    }

    pub fn reset_force(&mut self) {
        self.force = [0.0; 2];
    }
}